packages/ui/src/components/Select/use-model.ts

```diff
@@ -81,7 +88,26 @@ export function useSelectModel<T = any>({
     isValueSelected(value) ? unselect(value) : select(value)
   }
 
-  const selectedOptions = computed<SelectOption[]>(() => options.value.filter((x) => isValueSelected(x.value)))
+  function toggleGroupSelection(optionGroup: SelectOptionGroup<T>): void {
+    if (isGroupSelected(optionGroup)) {
+      const optionGroupSet = new Set(optionGroup.items.map((x) => x.value))
+      model.value = modelAsArray.value.filter((x) => !optionGroupSet.has(x))
+
+      return
+    }
+
+    model.value = [...new Set([...modelAsArray.value, ...optionGroup.items.map((x) => x.value)])]
+  }
```

Comment:
Wouldn't using `else` be more readable then empty `return`?

```suggestion
  function toggleGroupSelection(optionGroup: SelectOptionGroup<T>): void {
    if (isGroupSelected(optionGroup)) {
      const optionGroupSet = new Set(optionGroup.items.map((x) => x.value))
      model.value = modelAsArray.value.filter((x) => !optionGroupSet.has(x))
    } else {
      model.value = [...new Set([...modelAsArray.value, ...optionGroup.items.map((x) => x.value)])]
    }
  }
```